### PR TITLE
Sh fix a memleak for broken heredoc input and wrong memory access

### PR DESCRIPTION
**Pull request by masatake**
(Both fuzz and noise with VG=1 are passed.)

### CODE REVIEW

File: sh.c [hunk 1]
```diff
@@ -179,7 +184,7 @@ static void findShTags (void)
 			{
 				aliasFound = FALSE;
 				functionFound = FALSE;
-				++cp;
+				cp += (*cp == '\0')? 0: 1;
```
Comment (b4n) [inline sh.c]: I'd rather write
```c
if (*cp != '\0')
    ++cp;
```
than use a ternary with a no-op

**Comment by b4n** [comment]
Apart from that cosmetic detail, LGTM

**Comment by b4n** [comment]
(I re-ran one of the OSX Travis tests that stalled, all is good now)

**Comment by masatake** [comment]
@b4n, thank you. I reflected your comment to the change.

**Pull request merged**